Build failure: python3Packages.afdko

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `nixpkgs#python3Packages.afdko`

### Build log

```
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pypa-build-hook
Using pypaBuildPhase
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing pytest-check-hook
Using pytestCheckPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/52v9hha28zx91x4qxza3602alprh6jla-afdko-4.0.0.tar.gz
source root is afdko-4.0.0
setting SOURCE_DATE_EPOCH to timestamp 1694471944 of file afdko-4.0.0/setup.cfg
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
applying patch /nix/store/0icrfhj3wrwrvcvfl7kx09psligvrzqk-no-pypi-build-tools.patch
patching file setup.py
Hunk #1 succeeded at 198 (offset 2 lines).
applying patch /nix/store/00r3wz6naq0nmrg7jnf2bgsppkf1ia6f-use-dynamic-system-antlr4-runtime.patch
patching file CMakeLists.txt
Hunk #1 succeeded at 36 with fuzz 2.
patching file c/makeotf/lib/cffread/CMakeLists.txt
Hunk #1 succeeded at 8 with fuzz 1 (offset -5 lines).
patching file c/makeotf/lib/hotconv/CMakeLists.txt
Hunk #1 succeeded at 69 with fuzz 2.
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Executing pypaBuildPhase
Creating a wheel...
[1m* Getting build dependencies for wheel...[0m
/nix/store/2hkxasw90q7sp97jzqxnhwcv829xgklq-python3.11-setuptools-68.2.2/lib/python3.11/site-packages/setuptools/config/setupcfg.py:293: _DeprecatedConfig: Deprecated config in `setup.cfg`
!!

        ********************************************************************************
        The license_file parameter is deprecated, use license_files instead.

        This deprecation is overdue, please update your project and remove deprecated
        calls to avoid build errors in the future.

        See https://setuptools.pypa.io/en/latest/userguide/declarative_config.html for details.
        ********************************************************************************

!!
  parsed = self.parsers.get(option_name, lambda x: x)(value)
[1m* Building wheel...[0m
/nix/store/2hkxasw90q7sp97jzqxnhwcv829xgklq-python3.11-setuptools-68.2.2/lib/python3.11/site-packages/setuptools/config/setupcfg.py:293: _DeprecatedConfig: Deprecated config in `setup.cfg`
!!

        ********************************************************************************
        The license_file parameter is deprecated, use license_files instead.

        This deprecation is overdue, please update your project and remove deprecated
        calls to avoid build errors in the future.

        See https://setuptools.pypa.io/en/latest/userguide/declarative_config.html for details.
        ********************************************************************************

!!
  parsed = self.parsers.get(option_name, lambda x: x)(value)


--------------------------------------------------------------------------------
-- Trying 'Ninja' generator
--------------------------------
---------------------------
----------------------
-----------------
------------
-------
--
Not searching for unused variables given on the command line.
[0mCMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

[0m
sh: ps: command not found
sh: ps: command not found
sh: ps: command not found
-- The C compiler identification is Clang 16.0.6
sh: ps: command not found
-- Detecting C compiler ABI info
sh: ps: command not found
sh: ps: command not found
sh: ps: command not found
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/3fpl3hs6ii820yxp0rbafx5phnmj8ng7-clang-wrapper-16.0.6/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- The CXX compiler identification is Clang 16.0.6
sh: ps: command not found
-- Detecting CXX compiler ABI info
sh: ps: command not found
sh: ps: command not found
sh: ps: command not found
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/3fpl3hs6ii820yxp0rbafx5phnmj8ng7-clang-wrapper-16.0.6/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done (1.5s)
-- Generating done (0.0s)
-- Build files have been written to: /tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/_cmake_test_compile/build
--
-------
------------
-----------------
----------------------
---------------------------
--------------------------------
-- Trying 'Ninja' generator - success
--------------------------------------------------------------------------------

Configuring Project
  Working directory:
    /private/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/_skbuild/macosx-11.0-arm64-3.11/cmake-build
  Command:
    /nix/store/xckcwsyrgm5jwlwqgng19j8r0gy4w8ar-cmake-3.27.7/bin/cmake /private/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0 -G Ninja -DCMAKE_MAKE_PROGRAM:FILEPATH=/nix/store/r55dr1zqvhhq32axn0hgvyv9qv5c9wsf-ninja-1.11.1/bin/ninja --no-warn-unused-cli -DCMAKE_INSTALL_PREFIX:PATH=/private/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/_skbuild/macosx-11.0-arm64-3.11/cmake-install -DPYTHON_VERSION_STRING:STRING=3.11.6 -DSKBUILD:INTERNAL=TRUE -DCMAKE_MODULE_PATH:PATH=/nix/store/rjggdqc6xvcznjpds28rfzib6rpq2njw-python3.11-scikit-build-0.17.6/lib/python3.11/site-packages/skbuild/resources/cmake -DPYTHON_EXECUTABLE:PATH=/nix/store/ai4hd8f1xhr0rfjdr17bxx1rwi42sx97-python3-3.11.6/bin/python3.11 -DPYTHON_INCLUDE_DIR:PATH=/nix/store/ai4hd8f1xhr0rfjdr17bxx1rwi42sx97-python3-3.11.6/include/python3.11 -DPYTHON_LIBRARY:PATH=/nix/store/ai4hd8f1xhr0rfjdr17bxx1rwi42sx97-python3-3.11.6/lib/libpython3.11.dylib -DPython_EXECUTABLE:PATH=/nix/store/ai4hd8f1xhr0rfjdr17bxx1rwi42sx97-python3-3.11.6/bin/python3.11 -DPython_ROOT_DIR:PATH=/nix/store/ai4hd8f1xhr0rfjdr17bxx1rwi42sx97-python3-3.11.6 -DPython_FIND_REGISTRY:STRING=NEVER -DPython_INCLUDE_DIR:PATH=/nix/store/ai4hd8f1xhr0rfjdr17bxx1rwi42sx97-python3-3.11.6/include/python3.11 -DPython3_EXECUTABLE:PATH=/nix/store/ai4hd8f1xhr0rfjdr17bxx1rwi42sx97-python3-3.11.6/bin/python3.11 -DPython3_ROOT_DIR:PATH=/nix/store/ai4hd8f1xhr0rfjdr17bxx1rwi42sx97-python3-3.11.6 -DPython3_FIND_REGISTRY:STRING=NEVER -DPython3_INCLUDE_DIR:PATH=/nix/store/ai4hd8f1xhr0rfjdr17bxx1rwi42sx97-python3-3.11.6/include/python3.11 -DCMAKE_MAKE_PROGRAM:FILEPATH=/nix/store/r55dr1zqvhhq32axn0hgvyv9qv5c9wsf-ninja-1.11.1/bin/ninja -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=11.0 -DCMAKE_OSX_ARCHITECTURES:STRING=arm64 -DCMAKE_BUILD_TYPE:STRING=Release

Not searching for unused variables given on the command line.
sh: ps: command not found
sh: ps: command not found
sh: ps: command not found
-- The C compiler identification is Clang 16.0.6
sh: ps: command not found
-- The CXX compiler identification is Clang 16.0.6
sh: ps: command not found
-- Detecting C compiler ABI info
sh: ps: command not found
sh: ps: command not found
sh: ps: command not found
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/3fpl3hs6ii820yxp0rbafx5phnmj8ng7-clang-wrapper-16.0.6/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
sh: ps: command not found
sh: ps: command not found
sh: ps: command not found
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/3fpl3hs6ii820yxp0rbafx5phnmj8ng7-clang-wrapper-16.0.6/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Build type is Release
[33mCMake Warning (dev) at /nix/store/rjggdqc6xvcznjpds28rfzib6rpq2njw-python3.11-scikit-build-0.17.6/lib/python3.11/site-packages/skbuild/resources/cmake/FindPythonExtensions.cmake:245 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

Call Stack (most recent call first):
  CMakeLists.txt:25 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.
[0m
-- Found PythonInterp: /nix/store/ai4hd8f1xhr0rfjdr17bxx1rwi42sx97-python3-3.11.6/bin/python3.11 (found version "3.11.6") 
[33mCMake Warning (dev) at /nix/store/rjggdqc6xvcznjpds28rfzib6rpq2njw-python3.11-scikit-build-0.17.6/lib/python3.11/site-packages/skbuild/resources/cmake/FindPythonExtensions.cmake:252 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

Call Stack (most recent call first):
  CMakeLists.txt:25 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.
[0m
-- Found PythonLibs: /nix/store/ai4hd8f1xhr0rfjdr17bxx1rwi42sx97-python3-3.11.6/lib/libpython3.11.dylib (found version "3.11.6") 
<string>:2: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
<string>:2: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
-- Found LibXml2: /nix/store/yy7jhlyq2hhcgb47lzvygvjvckm906x9-libxml2-2.11.5/lib/libxml2.dylib (found version "2.11.5") 
-- Looking for floor in m
sh: ps: command not found
sh: ps: command not found
sh: ps: command not found
-- Looking for floor in m - found
sh: ps: command not found
Error: Could not run sw_vers:
No such file or directory
sh: ps: command not found
Error: Could not run sw_vers:
No such file or directory
sh: ps: command not found
Error: Could not run sw_vers:
No such file or directory
-- Found Python3: /nix/store/ai4hd8f1xhr0rfjdr17bxx1rwi42sx97-python3-3.11.6/bin/python3.11 (found version "3.11.6") found components: Interpreter 
sh: ps: command not found
-- Configuring done (1.5s)
-- Generating done (0.1s)
-- Build files have been written to: /tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/_skbuild/macosx-11.0-arm64-3.11/cmake-build

[0/209] Building C object c/shared/source/CMakeFiles/absfont.dir/absfont/absfont.c.o[K
[0/209] Building C object c/shared/source/CMakeFiles/absfont.dir/absfont/absfont_afm.c.o[K
[0/209] Building C object c/shared/source/CMakeFiles/absfont.dir/absfont/absfont_compare.c.o[K
[0/209] Building C object c/shared/source/CMakeFiles/absfont.dir/absfont/absfont_desc.c.o[K
[0/209] Building C object c/shared/source/CMakeFiles/absfont.dir/absfont/absfont_draw.c.o[K
[0/209] Building C object c/shared/source/CMakeFiles/absfont.dir/absfont/absfont_dump.c.o[K
[0/209] Building C object c/shared/source/CMakeFiles/absfont.dir/absfont/absfont_metrics.c.o[K
[0/209] Building C object c/shared/source/CMakeFiles/absfont.dir/absfont/absfont_path.c.o[K
[0/209] Building C object c/shared/source/CMakeFiles/cfembed.dir/cfembed/cfembed.c.o[K
[0/209] Building C object c/shared/source/CMakeFiles/cffread.dir/cffread/cffread.c.o[K
[1/209] Building C object c/shared/source/CMakeFiles/absfont.dir/absfont/absfont_afm.c.o[K
[1/209] Building C object c/shared/source/CMakeFiles/cffwrite.dir/cffwrite/cffwrite.c.o[K
[2/209] Building C object c/shared/source/CMakeFiles/absfont.dir/absfont/absfont_metrics.c.o[K
[2/209] Building C object c/shared/source/CMakeFiles/cffwrite.dir/cffwrite/cffwrite_charset.c.o[K
[3/209] Building C object c/shared/source/CMakeFiles/absfont.dir/absfont/absfont_desc.c.o[K
[3/209] Building C object c/shared/source/CMakeFiles/cffwrite.dir/cffwrite/cffwrite_dict.c.o[K
[4/209] Building C object c/shared/source/CMakeFiles/absfont.dir/absfont/absfont_compare.c.o[K
[4/209] Building C object c/shared/source/CMakeFiles/cffwrite.dir/cffwrite/cffwrite_encoding.c.o[K
[5/209] Building C object c/shared/source/CMakeFiles/absfont.dir/absfont/absfont_draw.c.o[K
[5/209] Building C object c/shared/source/CMakeFiles/cffwrite.dir/cffwrite/cffwrite_fdselect.c.o[K
[6/209] Building C object c/shared/source/CMakeFiles/absfont.dir/absfont/absfont.c.o[K
[6/209] Building C object c/shared/source/CMakeFiles/cffwrite.dir/cffwrite/cffwrite_sindex.c.o[K
[7/209] Building C object c/shared/source/CMakeFiles/absfont.dir/absfont/absfont_dump.c.o[K
/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/c/shared/source/absfont/absfont_dump.c:474:86: warning: implicit conversion from 'long' to 'float' changes value from 9223372036854775807 to 9223372036854775808 [-Wimplicit-const-int-float-conversion]
    if ((fabs(value - roundf(value)) < TX_EPSILON) && (value > LONG_MIN) && (value < LONG_MAX))
                                                                                   ~ ^~~~~~~~
/nix/store/3fpl3hs6ii820yxp0rbafx5phnmj8ng7-clang-wrapper-16.0.6/resource-root/include/limits.h:47:19: note: expanded from macro 'LONG_MAX'
#define LONG_MAX  __LONG_MAX__
                  ^~~~~~~~~~~~
<built-in>:48:22: note: expanded from macro '__LONG_MAX__'
#define __LONG_MAX__ 9223372036854775807L
                     ^~~~~~~~~~~~~~~~~~~~
1 warning generated.

[7/209] Building C object c/shared/source/CMakeFiles/cffwrite.dir/cffwrite/cffwrite_subr.c.o[K
[8/209] Building C object c/shared/source/CMakeFiles/cffwrite.dir/cffwrite/cffwrite_charset.c.o[K
[8/209] Building C object c/shared/source/CMakeFiles/cffwrite.dir/cffwrite/cffwrite_t2cstr.c.o[K
[9/209] Building C object c/shared/source/CMakeFiles/cffwrite.dir/cffwrite/cffwrite_encoding.c.o[K
[9/209] Building C object c/shared/source/CMakeFiles/cffwrite.dir/cffwrite/cffwrite_varstore.c.o[K
[10/209] Building C object c/shared/source/CMakeFiles/cffwrite.dir/cffwrite/cffwrite_fdselect.c.o[K
[10/209] Building C object c/shared/source/CMakeFiles/ctutil.dir/ctutil/ctutil.c.o[K
[11/209] Building C object c/shared/source/CMakeFiles/cffwrite.dir/cffwrite/cffwrite_sindex.c.o[K
[11/209] Building C object c/shared/source/CMakeFiles/dynarr.dir/dynarr/dynarr.c.o[K
[12/209] Building C object c/shared/source/CMakeFiles/cffwrite.dir/cffwrite/cffwrite_varstore.c.o[K
[12/209] Building C object c/shared/source/CMakeFiles/nameread.dir/nameread/nameread.c.o[K
[13/209] Building C object c/shared/source/CMakeFiles/ctutil.dir/ctutil/ctutil.c.o[K
[13/209] Linking C static library c/shared/source/libctutil.a[K
[14/209] Building C object c/shared/source/CMakeFiles/dynarr.dir/dynarr/dynarr.c.o[K
[14/209] Linking C static library c/shared/source/libdynarr.a[K
[15/209] Building C object c/shared/source/CMakeFiles/cffwrite.dir/cffwrite/cffwrite_dict.c.o[K
[15/209] Building C object c/shared/source/CMakeFiles/pdfwrite.dir/pdfwrite/pdfwrite.c.o[K
[16/209] Linking C static library c/shared/source/libdynarr.a[K
[16/209] Building C object c/shared/source/CMakeFiles/pstoken.dir/pstoken/pstoken.c.o[K
[17/209] Linking C static library c/shared/source/libctutil.a[K
[17/209] Building C object c/shared/source/CMakeFiles/sfntread.dir/sfntread/sfntread.c.o[K
[18/209] Building C object c/shared/source/CMakeFiles/nameread.dir/nameread/nameread.c.o[K
[18/209] Linking C static library c/shared/source/libnameread.a[K
[19/209] Building C object c/shared/source/CMakeFiles/cfembed.dir/cfembed/cfembed.c.o[K
[19/209] Linking C static library c/shared/source/libcfembed.a[K
[20/209] Building C object c/shared/source/CMakeFiles/sfntread.dir/sfntread/sfntread.c.o[K
[20/209] Linking C static library c/shared/source/libsfntread.a[K
[21/209] Linking C static library c/shared/source/libnameread.a[K
[21/209] Building C object c/shared/source/CMakeFiles/sfntwrite.dir/sfntwrite/sfntwrite.c.o[K
[22/209] Linking C static library c/shared/source/libcfembed.a[K
[22/209] Building C object c/shared/source/CMakeFiles/sha1.dir/sha1/sha1.c.o[K
[23/209] Linking C static library c/shared/source/libsfntread.a[K
[23/209] Building C object c/shared/source/CMakeFiles/support.dir/support/canthappen.c.o[K
[24/209] Building C object c/shared/source/CMakeFiles/pstoken.dir/pstoken/pstoken.c.o[K
[24/209] Linking C static library c/shared/source/libpstoken.a[K
[25/209] Building C object c/shared/source/CMakeFiles/pdfwrite.dir/pdfwrite/pdfwrite.c.o[K
[25/209] Linking C static library c/shared/source/libpdfwrite.a[K
[26/209] Building C object c/shared/source/CMakeFiles/support.dir/support/canthappen.c.o[K
[26/209] Building C object c/shared/source/CMakeFiles/support.dir/support/except.c.o[K
[27/209] Building C object c/shared/source/CMakeFiles/sfntwrite.dir/sfntwrite/sfntwrite.c.o[K
[27/209] Linking C static library c/shared/source/libsfntwrite.a[K
[28/209] Linking C static library c/shared/source/libpstoken.a[K
[28/209] Building C object c/shared/source/CMakeFiles/support.dir/support/fixed.c.o[K
[29/209] Linking C static library c/shared/source/libpdfwrite.a[K
[29/209] Building C object c/shared/source/CMakeFiles/svgwrite.dir/svgwrite/svgwrite.c.o[K
[30/209] Building C object c/shared/source/CMakeFiles/sha1.dir/sha1/sha1.c.o[K
[30/209] Linking C static library c/shared/source/libsha1.a[K
[31/209] Building C object c/shared/source/CMakeFiles/support.dir/support/except.c.o[K
[31/209] Building C object c/shared/source/CMakeFiles/svread.dir/svread/svread.c.o[K
[32/209] Building C object c/shared/source/CMakeFiles/support.dir/support/fixed.c.o[K
[32/209] Linking C static library c/shared/source/libsupport.a[K
[33/209] Linking C static library c/shared/source/libsfntwrite.a[K
[33/209] Building C object c/shared/source/CMakeFiles/t1cstr.dir/t1cstr/t1cstr.c.o[K
[34/209] Linking C static library c/shared/source/libsha1.a[K
[34/209] Building C object c/shared/source/CMakeFiles/t1read.dir/t1read/t1read.c.o[K
[35/209] Linking C static library c/shared/source/libsupport.a[K
[35/209] Building C object c/shared/source/CMakeFiles/t1write.dir/t1write/t1write.c.o[K
[36/209] Building C object c/shared/source/CMakeFiles/svgwrite.dir/svgwrite/svgwrite.c.o[K
/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/c/shared/source/svgwrite/svgwrite.c:200:86: warning: implicit conversion from 'long' to 'float' changes value from 9223372036854775807 to 9223372036854775808 [-Wimplicit-const-int-float-conversion]
    if ((fabs(value - roundf(value)) < TX_EPSILON) && (value > LONG_MIN) && (value < LONG_MAX))
                                                                                   ~ ^~~~~~~~
/nix/store/3fpl3hs6ii820yxp0rbafx5phnmj8ng7-clang-wrapper-16.0.6/resource-root/include/limits.h:47:19: note: expanded from macro 'LONG_MAX'
#define LONG_MAX  __LONG_MAX__
                  ^~~~~~~~~~~~
<built-in>:48:22: note: expanded from macro '__LONG_MAX__'
#define __LONG_MAX__ 9223372036854775807L
                     ^~~~~~~~~~~~~~~~~~~~
1 warning generated.

[36/209] Linking C static library c/shared/source/libsvgwrite.a[K
[37/209] Linking C static library c/shared/source/libsvgwrite.a[K
[37/209] Building C object c/shared/source/CMakeFiles/t2cstr.dir/t2cstr/t2cstr.c.o[K
[38/209] Building C object c/shared/source/CMakeFiles/cffwrite.dir/cffwrite/cffwrite.c.o[K
[38/209] Building C object c/shared/source/CMakeFiles/ttread.dir/ttread/ttread.c.o[K
[39/209] Building C object c/shared/source/CMakeFiles/svread.dir/svread/svread.c.o[K
[39/209] Linking C static library c/shared/source/libsvread.a[K
[40/209] Building C object c/shared/source/CMakeFiles/t1cstr.dir/t1cstr/t1cstr.c.o[K
[40/209] Linking C static library c/shared/source/libt1cstr.a[K
[41/209] Building C object c/shared/source/CMakeFiles/cffread.dir/cffread/cffread.c.o[K
[41/209] Linking C static library c/shared/source/libcffread.a[K
[42/209] Linking C static library c/shared/source/libsvread.a[K
[42/209] Building C object c/shared/source/CMakeFiles/tx_shared.dir/tx_shared/tx_shared.c.o[K
[43/209] Linking C static library c/shared/source/libt1cstr.a[K
[43/209] Building C object c/shared/source/CMakeFiles/uforead.dir/uforead/uforead.c.o[K
[44/209] Linking C static library c/shared/source/libcffread.a[K
[44/209] Building C object c/shared/source/CMakeFiles/ufowrite.dir/ufowrite/ufowrite.c.o[K
[45/209] Building C object c/shared/source/CMakeFiles/absfont.dir/absfont/absfont_path.c.o[K
[45/209] Linking C static library c/shared/source/libabsfont.a[K
[46/209] Building C object c/shared/source/CMakeFiles/tx_shared.dir/tx_shared/tx_shared.c.o[K
[31mFAILED: [0mc/shared/source/CMakeFiles/tx_shared.dir/tx_shared/tx_shared.c.o 
/nix/store/3fpl3hs6ii820yxp0rbafx5phnmj8ng7-clang-wrapper-16.0.6/bin/clang  -I/nix/store/5qsssc7jh0ffgw86difvpi3i24kd8pj0-libxml2-2.11.5-dev/include/libxml2 -I/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/c/shared/source/../include -I/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/c/shared/source/../resource -O3 -DNDEBUG -arch arm64 -mmacosx-version-min=11.0 -MD -MT c/shared/source/CMakeFiles/tx_shared.dir/tx_shared/tx_shared.c.o -MF c/shared/source/CMakeFiles/tx_shared.dir/tx_shared/tx_shared.c.o.d -o c/shared/source/CMakeFiles/tx_shared.dir/tx_shared/tx_shared.c.o -c /tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/c/shared/source/tx_shared/tx_shared.c
/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/c/shared/source/tx_shared/tx_shared.c:526:20: error: incompatible function pointer types assigning to 'size_t (*)(ctlStreamCallbacks *, void *, char **)' (aka 'unsigned long (*)(struct ctlStreamCallbacks_ *, void *, char **)') from 'size_t (ctlStreamCallbacks *, Stream *, char **)' (aka 'unsigned long (struct ctlStreamCallbacks_ *, Stream *, char **)') [-Wincompatible-function-pointer-types]
    h->cb.stm.read = stm_read;
                   ^ ~~~~~~~~
/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/c/shared/source/tx_shared/tx_shared.c:527:24: error: incompatible function pointer types assigning to 'size_t (*)(ctlStreamCallbacks *, void *, xmlDocPtr *)' (aka 'unsigned long (*)(struct ctlStreamCallbacks_ *, void *, struct _xmlDoc **)') from 'size_t (ctlStreamCallbacks *, Stream *, xmlDocPtr *)' (aka 'unsigned long (struct ctlStreamCallbacks_ *, Stream *, struct _xmlDoc **)') [-Wincompatible-function-pointer-types]
    h->cb.stm.xml_read = stm_xml_read;
                       ^ ~~~~~~~~~~~~
2 errors generated.

[47/209] Building C object c/shared/source/CMakeFiles/uforead.dir/uforead/uforead.c.o[K
[31mFAILED: [0mc/shared/source/CMakeFiles/uforead.dir/uforead/uforead.c.o 
/nix/store/3fpl3hs6ii820yxp0rbafx5phnmj8ng7-clang-wrapper-16.0.6/bin/clang  -I/nix/store/5qsssc7jh0ffgw86difvpi3i24kd8pj0-libxml2-2.11.5-dev/include/libxml2 -I/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/c/shared/source/../include -I/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/c/shared/source/../resource -O3 -DNDEBUG -arch arm64 -mmacosx-version-min=11.0 -MD -MT c/shared/source/CMakeFiles/uforead.dir/uforead/uforead.c.o -MF c/shared/source/CMakeFiles/uforead.dir/uforead/uforead.c.o.d -o c/shared/source/CMakeFiles/uforead.dir/uforead/uforead.c.o -c /tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/c/shared/source/uforead/uforead.c
/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/c/shared/source/uforead/uforead.c:1243:20: error: incompatible integer to pointer conversion passing 'int' to parameter of type 'char *' [-Wint-conversion]
        message(h, ufoErrParse, "Encountered glyph reference %s in alternate layer's contents.plist with an empty file path. ", glyphName);
                   ^~~~~~~~~~~
/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/c/shared/source/uforead/uforead.c:396:47: note: passing argument to parameter 'fmt' here
static void CTL_CDECL message(ufoCtx h, char* fmt, ...) {
                                              ^
/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/c/shared/source/uforead/uforead.c:2004:16: error: incompatible pointer to integer conversion returning 'void *' from a function with result type 'long' [-Wint-conversion]
        return NULL;
               ^~~~
/nix/store/3fpl3hs6ii820yxp0rbafx5phnmj8ng7-clang-wrapper-16.0.6/resource-root/include/stddef.h:89:16: note: expanded from macro 'NULL'
#  define NULL ((void*)0)
               ^~~~~~~~~~
/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/c/shared/source/uforead/uforead.c:2019:16: error: incompatible pointer to integer conversion returning 'void *' from a function with result type 'unsigned long' [-Wint-conversion]
        return NULL;
               ^~~~
/nix/store/3fpl3hs6ii820yxp0rbafx5phnmj8ng7-clang-wrapper-16.0.6/resource-root/include/stddef.h:89:16: note: expanded from macro 'NULL'
#  define NULL ((void*)0)
               ^~~~~~~~~~
3 errors generated.

[48/209] Linking C static library c/shared/source/libabsfont.a[K
[49/209] Building C object c/shared/source/CMakeFiles/cffwrite.dir/cffwrite/cffwrite_subr.c.o[K
[50/209] Building C object c/shared/source/CMakeFiles/t2cstr.dir/t2cstr/t2cstr.c.o[K
[51/209] Building C object c/shared/source/CMakeFiles/ufowrite.dir/ufowrite/ufowrite.c.o[K
/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/c/shared/source/ufowrite/ufowrite.c:228:83: warning: implicit conversion from 'long' to 'float' changes value from 9223372036854775807 to 9223372036854775808 [-Wimplicit-const-int-float-conversion]
    if ((fabs(value - roundf(value)) < TX_EPSILON) && value > LONG_MIN && value < LONG_MAX)
                                                                                ~ ^~~~~~~~
/nix/store/3fpl3hs6ii820yxp0rbafx5phnmj8ng7-clang-wrapper-16.0.6/resource-root/include/limits.h:47:19: note: expanded from macro 'LONG_MAX'
#define LONG_MAX  __LONG_MAX__
                  ^~~~~~~~~~~~
<built-in>:48:22: note: expanded from macro '__LONG_MAX__'
#define __LONG_MAX__ 9223372036854775807L
                     ^~~~~~~~~~~~~~~~~~~~
/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/c/shared/source/ufowrite/ufowrite.c:629:38: warning: floating-point comparison is always true; constant cannot be represented exactly in type 'float' [-Wliteral-range]
    if (privateDict->ExpansionFactor != cff_DFLT_ExpansionFactor){
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~
2 warnings generated.

[52/209] Building C object c/shared/source/CMakeFiles/cffwrite.dir/cffwrite/cffwrite_t2cstr.c.o[K
[53/209] Building C object c/shared/source/CMakeFiles/t1write.dir/t1write/t1write.c.o[K
/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/c/shared/source/t1write/t1write.c:348:28: warning: incompatible pointer types passing 'uint16_t *' (aka 'unsigned short *') to parameter of type 'unsigned char *' [-Wincompatible-pointer-types]
        if (writeTmp(h, 1, &info->iFD))
                           ^~~~~~~~~~
/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/c/shared/source/t1write/t1write.c:327:56: note: passing argument to parameter 'ptr' here
static int writeTmp(t1wCtx h, long cnt, unsigned char *ptr) {
                                                       ^
1 warning generated.

[54/209] Building C object c/shared/source/CMakeFiles/t1read.dir/t1read/t1read.c.o[K
/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/c/shared/source/t1read/t1read.c:348:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]
}
^
1 warning generated.

[55/209] Building C object c/shared/source/CMakeFiles/ttread.dir/ttread/ttread.c.o[K
ninja: build stopped: subcommand failed.
Traceback (most recent call last):
  File "/nix/store/rjggdqc6xvcznjpds28rfzib6rpq2njw-python3.11-scikit-build-0.17.6/lib/python3.11/site-packages/skbuild/setuptools_wrap.py", line 674, in setup
    cmkr.make(make_args, install_target=cmake_install_target, env=env)
  File "/nix/store/rjggdqc6xvcznjpds28rfzib6rpq2njw-python3.11-scikit-build-0.17.6/lib/python3.11/site-packages/skbuild/cmaker.py", line 697, in make
    self.make_impl(clargs=clargs, config=config, source_dir=source_dir, install_target=install_target, env=env)
  File "/nix/store/rjggdqc6xvcznjpds28rfzib6rpq2njw-python3.11-scikit-build-0.17.6/lib/python3.11/site-packages/skbuild/cmaker.py", line 742, in make_impl
    raise SKBuildError(msg)

An error occurred while building with CMake.
  Command:
    /nix/store/xckcwsyrgm5jwlwqgng19j8r0gy4w8ar-cmake-3.27.7/bin/cmake --build . --target install --config Release --
  Install target:
    install
  Source directory:
    /private/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0
  Working directory:
    /private/tmp/nix-build-python3.11-afdko-4.0.0.drv-0/afdko-4.0.0/_skbuild/macosx-11.0-arm64-3.11/cmake-build
Please check the install target is valid and see CMake's output for more information.


[91mERROR[0m Backend subprocess exited when trying to invoke build_wheel

```

### Additional context

Add any other context about the problem here.

### Notify maintainers
@sternenseemann
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
~ via C v15.0.0-clang via λ 9.4.7 via 🌙 via  v18.18.2
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.1.0, macOS 14.1.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixpkgs"`
 - channels(rocko): `"nixpkgs, home-manager"`
 - nixpkgs: `/Users/rocko/.nix-defexpr/channels/nixpkgs`
```

### Priorities

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/afdko/default.nix
+++ b/pkgs/development/python-modules/afdko/default.nix
@@ -64,6 +64,11 @@ buildPythonPackage rec {
     ./use-dynamic-system-antlr4-runtime.patch
   ];
 
+  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isClang (toString [
+    "-Wno-error=incompatible-function-pointer-types"
+    "-Wno-error=int-conversion"
+  ]);
+
   # setup.py will always (re-)execute cmake in buildPhase
   dontConfigure = true;
 

